presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java

```diff
@@ -107,6 +108,7 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
             }
 
             // authentication succeeded
+            authenticator.extractUser(principal).ifPresent(extractedUser -> request.setAttribute(PRESTO_USER, extractedUser));
```

Comment:
what about setting the attribute unconditionally:
```
request.setAttribute(PRESTO_USER, authenticator.extractUser(principal));
```
?

(of course, the type becomes `Optional<String>` now)

Reply:
Given how servlets work, I'd still have to test for null on the receiver side in case this filter wasn't called (just to be safe), so I think that would make the receiver more complex

Reply:
is "this filter wasn't called" an expected situation?